drivers/SmartThings/matter-lock/src/new-matter-lock/init.lua

```diff
@@ -0,0 +1,1777 @@
+  local res = device.MATTER_DEFAULT_ENDPOINT
+  local eps = device:get_endpoints(cluster)
+  table.sort(eps)
+  for _, v in ipairs(eps) do
+    if v ~= 0 then --0 is the matter RootNode endpoint
+      return v
+    end
+  end
+  device.log.warn(string.format("Did not find default endpoint, will use endpoint %d instead", device.MATTER_DEFAULT_ENDPOINT))
+  return res
+end
+
+local function component_to_endpoint(device, component_name)
+  return find_default_endpoint(device, clusters.DoorLock.ID)
+end
+
+local function device_init(driver, device)
+  log.info_with({hub_logs=true}, string.format("!!!!!!!!!!!!!!! device_added !!!!!!!!!!!!!")) -- needs to remove
+  device:set_component_to_endpoint_fn(component_to_endpoint)
+  device:subscribe()
+ end
+
+local function device_added(driver, device)
+  device:emit_event(capabilities.lockAlarm.alarm.clear({state_change = true}))
+end
+
+local function do_configure(driver, device)
+  log.info_with({hub_logs=true}, string.format("!!!!!!!!!!!!!!! do_configure !!!!!!!!!!!!!")) -- needs to remove
+
+  local user_eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.Feature.USER})
+  local pin_eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.Feature.PIN_CREDENTIAL})
+  local week_schedule_eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.Feature.WEEK_DAY_ACCESS_SCHEDULES})
+  local year_schedule_eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.Feature.YEAR_DAY_ACCESS_SCHEDULES})
+
+  local profile_name = "lock"
+  if #user_eps > 0 then
+    profile_name = profile_name .. "-user"
+    if #pin_eps > 0 then
+      profile_name = profile_name .. "-pin"
+    end
+    if #week_schedule_eps + #year_schedule_eps > 0 then
+      profile_name = profile_name .. "-schedule"
+    end
+  else
+    profile_name = "base-lock"
+  end
+  device.log.info_with({hub_logs=true}, string.format("Updating device profile to %s.", profile_name))
+  device:try_update_metadata({profile = profile_name})
+end
+
+local function info_changed(driver, device, event, args)
+  for cap_id, attributes in pairs(subscribed_attributes) do
+    if device:supports_capability_by_id(cap_id) then
+      for _, attr in ipairs(attributes) do
+        device:add_subscribed_attribute(attr)
+      end
+    end
+  end
+  for cap_id, events in pairs(subscribed_events) do
+    if device:supports_capability_by_id(cap_id) then
+      for _, e in ipairs(events) do
+        device:add_subscribed_event(e)
+      end
+    end
+  end
+  device:subscribe()
+end
+
+-- Matter Handler
+----------------
+-- Lock State --
+----------------
+local function lock_state_handler(driver, device, ib, response)
+  log.info_with({hub_logs=true}, string.format("!!!!!!!!!!!!!!! lock_state_handler !!!!!!!!!!!!!")) -- needs to remove
+  local LockState = DoorLock.attributes.LockState
+  local attr = capabilities.lock.lock
+  local LOCK_STATE = {
+    [LockState.NOT_FULLY_LOCKED] = attr.unknown(),
```

Comment:
This code has recently changed upstream. Please make sure to incorporate those changes into this code:

https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1595
https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1602

Reply:
Thank you for your review 👍